### PR TITLE
Change registry

### LINKED ISSUE #23: replace personal registry
replace `ashnam` docker hub registry with another registry (like [GHCR](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry))

### PR DESCRIPTION
**Pull request by ashnamehrotra**

### CODE REVIEW

File: .github/workflows/build-eraser.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Build and Push Eraser Image
```
Comment (sozercan) [inline .github/workflows/build-eraser.yaml]: let's call this manager (change filename too)

File: controllers/imagelist/imagelist_controller.go [hunk 2]
```diff
@@ -90,8 +90,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 						RestartPolicy: "Never",
 						Containers: []corev1.Container{
 							{
-								Name:            "remove-images",
-								Image:           "aldaircoronel/remove_images:latest",
+								Name:            "eraser",
+								Image:           "ghcr.io/azure/eraser:latest",
```
Comment (sozercan) [inline controllers/imagelist/imagelist_controller.go:94]: this doesn't exist. @AldairCoronel with the e2e PR we should be able to override this.

**Review by sozercan — APPROVED** [review]
LGTM

File: Makefile [hunk 3]
```diff
@@ -1,6 +1,7 @@
 
 # Image URL to use all building/pushing image targets
-IMG ?= aldaircoronel/controller:latest
+IMG ?= ghcr.io/azure/eraser-manager:latest
```
Comment (cpuguy83) [inline Makefile:3]: Did we want this to be something other than latest?

```suggestion
COMMIT := $(shell git rev-parse --short HEAD)
DIRTY := $(shell [ -n "$(git status --porcelain)" ] && echo "-dirty")
IMG ?= ghcr.io/azure/eraser-manager:$(COMMIT)$(DIRTY)
```

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      - name: Log in to the GHCR
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish
+        run: |
+          make docker-push IMG=${{ env.REGISTRY }}/${{ env.MANAGER_IMAGE_REPO }}:${TAG}
```
Comment (cpuguy83) [inline .github/workflows/release.yaml:69]: Note that you can add `--push` to the build command.

**Review by cpuguy83 — APPROVED** [review]
Minor comment on image naming, otherwise LGTM

File: Makefile (hunk 3 above)
Comment (sozercan) [reply Makefile:3]: this will get overridden with `IMG` value (including tag) when a release is tagged

File: .github/workflows/release.yaml (hunk 4 above)
Comment (sozercan) [reply .github/workflows/release.yaml:69]: Yes, I wanted to make sure both images are built before pushing so if one of them fails to build we fail

Comment (cpuguy83) [reply .github/workflows/release.yaml:69]: Ah yes that makes sense.

File: Makefile (hunk 3 above)
Comment (sozercan) [reply Makefile:3]: example: https://github.com/sozercan/eraser/runs/3435922240?check_suite_focus=true

**Pull request merged**